storm-client/src/jvm/org/apache/storm/metrics2/TaskMetrics.java

```diff
@@ -50,6 +52,13 @@ public TaskMetrics(WorkerTopologyContext context, String componentId, Integer ta
         this.samplingRate = ConfigUtils.samplingRate(topoConf);
     }
 
+    public void setCapacity(double capacity) {
```

Comment:
Does this method need to guard against multiple calls, by tracking previous capacity and adding only the incremental difference to gauge.addValue()  - so the total capacity is always the capacity in the latest call to setCapacity? 

Alternatively, respecting only first call and ignoring subsequent calls.